src/sentry/integrations/notifications.py

```diff
@@ -0,0 +1,29 @@
+import logging
+
+from sentry.models import ExternalActor, NotificationSetting, Team
+from sentry.types.integrations import ExternalProviders
+
+logger = logging.getLogger("sentry.integrations.notifications")
+
+
+class NotifyBasicMixin:
+    def send_message(self, channel_id: str, message: str) -> None:
+        """
+        Send a message through the integration.
+        """
+        raise NotImplementedError
+
+    def notify_remove_external_team(self, external_team: ExternalActor, team: Team) -> None:
+        """
+        Notify through the integration that an external team has been removed.
+        """
+        raise NotImplementedError
+
+    def remove_notification_settings(self, actor_id: int, provider: str) -> None:
```

Comment:
The parameter `provider` is an `int` here. I think you should actually take the `provider` as an `ExternalProviders` enum and make the conversion from int in the model.

Reply:
Fixed, but @mgaeta it just occured to me that this function was being run on the model before and now only runs if the installation has the `NotifyBasicMixin` applied (otherwise we use the noop). Do you know if there are other integrations that need this functionality or is it limited to slack?

Reply:
If it's just slack, then this PR will work fine, I'm just not sure as to what NotificationSettings actually contains and how relevant it is to other integrations